airflow/providers/amazon/aws/operators/athena.py

```diff
@@ -28,9 +28,6 @@ class AWSAthenaOperator(BaseOperator):
     """
     An operator that submit presto query to athena.
 
-    If ``do_xcom_push`` is True, the QueryExecutionID assigned to the
-    query will be pushed to an XCom when it successfuly completes.
```

Comment:
I think this is still valid because:
https://github.com/apache/airflow/blob/7a5441836bbc8bbcb0b02bf68dbd93d63fe4ec6a/airflow/models/taskinstance.py#L1035-L1037

Reply:
If operator's `execute` method returns a `result` it doesn't mean that it will be always saved in XCom table. This will happen only if operator is used with `do_xcom_push=True`

Reply:
Got it, that makes sense. This behavior seems to correspond to the super classes and not to this specific class, hence why I don't think it needs to be mentioned in the docstring. Does that make sense?

Reply:
[Here is the setting in `BaseOperator` on `master`](https://github.com/apache/airflow/blob/eb403beea2d1035635b7bea24c49b6b964313e51/airflow/models/baseoperator.py#L359). Since operator docstrings usually do not mention super class functionality, I think this could be removed. The behavior should be evident by looking at the super class and understanding that `query_execution_id` is the value in question that will or will not be xcom_pushed.

Reply:
> Since operator docstrings usually do not mention super class functionality, I think this could be removed.

With that I can agree. My point was that this part of the PR description is not true:

>Since query_execution_id is now returned by execute, it will be xcom
pushed by default

Reply:
Noted! Just updated the PR description and title to reflect the full context. Thanks for pointing that out!